in 24.4 + java21 some demos do not run

### Description of the bug

This error is seen when running an app in java21, but it works in java17
```
Caused by: java.lang.IllegalArgumentException: Unsupported class file major version 65
    at org.objectweb.asm.ClassReader.<init> (ClassReader.java:199)
    at org.objectweb.asm.ClassReader.<init> (ClassReader.java:180)
    at org.objectweb.asm.ClassReader.<init> (ClassReader.java:166)
    at org.objectweb.asm.ClassReader.<init> (ClassReader.java:287)
    at org.eclipse.jetty.annotations.AnnotationParser.scanClass (AnnotationParser.java:896)
```

### Expected behavior

it works

### Minimal reproducible example

```
java -version (check that it is 21)
git clone git@github.com:vaadin/bookstore-example.git
cd bookstore-example
mvn
```

### Versions

- Vaadin / Flow version: 24.4.1 24.4.2
- Java version: JDK 21

--- a/pom.xml
+++ b/pom.xml
@@ -133,7 +133,7 @@
             <plugin>
                 <groupId>org.eclipse.jetty</groupId>
                 <artifactId>jetty-maven-plugin</artifactId>
-                <version>11.0.13</version>
+                <version>11.0.15</version>
                 <configuration>
                     <scan>2</scan>
                 </configuration>

